sklearn/utils/stats.py

```diff
@@ -15,47 +154,70 @@ def _weighted_percentile(array, sample_weight, percentile=50):
 
     Parameters
     ----------
-    array : 1D or 2D array
+    array : ndarray of shape (n,) or (n, m)
         Values to take the weighted percentile of.
 
-    sample_weight: 1D or 2D array
+    sample_weight: ndarray of shape (n,) or (n, m)
         Weights for each value in `array`. Must be same shape as `array` or
         of shape `(array.shape[0],)`.
 
-    percentile: int, default=50
+    percentile: int or float, default=50
```

Comment:
then it's technically just float, isn't it? usually when we say int or float, they mean different things, I think.